### ISSUE TITLE
Better logarithmic colorscale example

### ISSUE THREAD

**Issue by chriddyp** [plotly/documentation#1611]
Our current example does a nice job of scaling the colors but the labels in the colorbar are a little too condensed:
![image](https://user-images.githubusercontent.com/1280389/69890743-717b8880-12c5-11ea-90ad-6a45dd33c062.png)

I recognize that this may be a limitation of our colorscale axis (there is no `log` type), but could we have another example that "fakes" it by:
1. Setting custom `ticktext` and `tickvals`
2. Normalizing the `z` data if necessary
3. Remapping the hover data so be the un-normalized values

The solutions team (cc @ycaokris ) has been mapping data to color with logarithmic values lately, which is why I'm thinking it would be helpful to have a better recipe in our docs. Here are some colorscales I've seen from some sample apps:
![image](https://user-images.githubusercontent.com/1280389/69890791-c9b28a80-12c5-11ea-810a-8b639c50509a.png)
![image](https://user-images.githubusercontent.com/1280389/69890794-d1722f00-12c5-11ea-94a1-664090468710.png)

The logarithmic colorscales that I'm thinking about are more like this:
![image](https://user-images.githubusercontent.com/1280389/69890825-041c2780-12c6-11ea-823c-9920d9431f26.png)

Once this is updated, we should provide a backlink to the documentation by commenting in the Stack Overflow question about this topic: https://stackoverflow.com/questions/50392372/logarithmic-color-scale-in-plotly

cc @Mahdis-z @emmanuelle @nicolaskruchten

**Comment by rcwhitney** [comment]
Happy to see this is being brought up.  I'm currently working on a project where we are visualizing data with heatmaps and want the z-axis (color) to be logarithmic.

However it appears that the z-axis cannot be set to logarithmic for heatmaps (but haven't been able to 100% confirm this).  What we're seeing is when we set the tick marks to be logarithmic for the colorbar is exactly like the first example - and this sounds like the correct behavior if the data being displayed itself is not logarithmic.  If the data was, then the spacing between the log tick marks should be the same. 

I believe the example used above and in the documention is not displaying the z-axis logarithmically, hence you get that weirdness with the colorbar.    Still digging for more information though.

**Comment by nicolaskruchten** [comment]
So our docs actually have a couple different examples of ways to do this now: https://plotly.com/python/colorscales/#customizing-tick-text-on-logarithmic-color-bars and https://plotly.com/python/datashader/#exploring-correlations-of-a-large-dataset

**Comment by rcwhitney** [comment]
Yes! Those color bars look correct. 

May be best to switch out the example with one of those, since it appears density plots don't support logarithmic z-axises.

**Comment by nicolaskruchten** [comment]
You're right that we don't support log color axes per se: if you look at how these are done, they're linear color axes with log-scaled inputs, and then we change the tick labels from `N` to `1eN` basically.

**Comment by hhaensel** [comment]
I played a bit with the different examples, but as far as I can understand it the colors appear stretched and the bottom color is almost not recognisable.
I would be nice to have a way of setting colormapping to logarithmic.
Meanwhile I propose a different workaround, which has accurate logarithmic color-mapping:
```python
import numpy as np
import plotly.graph_objects as go

a = np.exp(np.array([[1,2,3],[4,5,6],[7,8,9]]))

def colorbar(zmin, zmax, n = 6):
    return dict(
        title = "Log color scale",
        tickmode = "array",
        tickvals = np.linspace(np.log10(zmin), np.log10(zmax), n),
        ticktext = np.round(10 ** np.linspace(np.log10(zmin), np.log10(zmax), n), 1)
    )

zmin = 1
zmax = 10000
go.Figure(go.Heatmap(z = np.log10(a), text = a,
    hovertemplate = 
    "<i>Logplot Demo</i><br>" +
    "<b>X</b>: %{x}<br>" +
    "<b>Y</b>: %{y}<br>" +
    "<b>Z</b>: %{text:.1f}",
    name = "",
    colorscale = "Portland",
    zmin = np.log10(zmin), zmax = np.log10(zmax),
    colorbar = colorbar(zmin, zmax, 5)
))
```
![image](https://user-images.githubusercontent.com/31985040/158620315-356489ab-9603-4189-869a-a87f3b2eb5cf.png)
